### PR TITLE
Add settlement finality latency metric

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

This PR adds a new metric to track transaction settlement finality. This is more accurate because it only tracks the case where a transaction is driven from raw transaction to effects with one try, successfully.
Also use custom buckets and built-in histogram as it's more accurate.

## Test Plan 

CI.

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/quorum_driver/mod.rs [hunk 1]
```diff
@@ -736,6 +738,15 @@ where
                 return;
             }
         };
+        quorum_driver
+            .metrics
+            .settlement_finality_latency
+            .with_label_values(&[if is_single_writer_tx {
+                "single_writer"
+            } else {
+                "shared"
```
Comment (longbowlu) [inline crates/sui-core/src/quorum_driver/mod.rs:747]: use TX_TYPE_SINGLE_WRITER_TX and TX_TYPE_SHARED_OBJ_TX in mysten-metrics

**Review by longbowlu — APPROVED** [review]
the caveat of this is when there are QD retries (e.g. network connection issues with validators), it is not captured in this metrics. So when get a high level view, we want to look at both this metrics and # of QD attempts to finality.

**Comment by lxfind** [comment]
> the caveat of this is when there are QD retries (e.g. network connection issues with validators), it is not captured in this metrics.

Can you elaborate on this more? Even when QD retries, this metric should still capture a successful finality metric (that exclude retry latency)?

**Comment by longbowlu** [comment]
> > the caveat of this is when there are QD retries (e.g. network connection issues with validators), it is not captured in this metrics.
> 
> Can you elaborate on this more? Even when QD retries, this metric should still capture a successful finality metric (that exclude retry latency)?

say it takes 10 times for a tx to finalize, the first few ones fail because of network issues or network overload. This metrics only covers the final successful one, however user/client perceived latency could be much higher

**Pull request merged**